clang/lib/CodeGen/CGCall.cpp

```diff
@@ -3507,6 +3507,9 @@ static llvm::StoreInst *findDominatingStoreToReturnValue(CodeGenFunction &CGF) {
       return nullptr;
     // These aren't actually possible for non-coerced returns, and we
     // only care about non-coerced returns on this code path.
+    // All memory instructions inside __try block are volatile.
```

Comment:
Functionally, this is adjusting the conditions of the assert, can you move the change into the assert? Essentially, make the assert be:
```
  assert(!SI->isAtomic() && (!SI->isVolatile() || CGF.currentFunctionUsesSEHTry()));
```